tests/ci/integration/openvpn_patch/aws-lc-openvpn2-6-x.patch

```diff
@@ -38,10 +38,22 @@ index c9fa7196..a48ef391 100644
  #endif
  
 diff --git a/src/openvpn/ssl_openssl.c b/src/openvpn/ssl_openssl.c
-index 2595f878..cf99c3ec 100644
+index 23e76231..cab58ff4 100644
 --- a/src/openvpn/ssl_openssl.c
 +++ b/src/openvpn/ssl_openssl.c
-@@ -1658,8 +1658,10 @@ tls_ctx_load_ca(struct tls_root_ctx *ctx, const char *ca_file,
+@@ -1431,7 +1431,11 @@ tls_ctx_use_external_ec_key(struct tls_root_ctx *ctx, EVP_PKEY *pkey)
+ 
+     /* Among init methods, we only need the finish method */
+     EC_KEY_METHOD_set_init(ec_method, NULL, openvpn_extkey_ec_finish, NULL, NULL, NULL, NULL);
++#ifdef OPENSSL_IS_AWSLC
++    EC_KEY_METHOD_set_sign(ec_method, ecdsa_sign, NULL, ecdsa_sign_sig);
++#else
+     EC_KEY_METHOD_set_sign(ec_method, ecdsa_sign, ecdsa_sign_setup, ecdsa_sign_sig);
++#endif
```

Comment:
How does our lack of support for "sign_setup" not end up breaking OpenVPN's usage here?

Reply:
Their sign_setup function, ecdsa_sign_setup, does nothing (returns 1). We want to upstream this change so they stop setting that field altogether. In OpenSSL, the only way this custom sign_setup functionality is invoked is 
1) Through ECDSA_sign_setup (which we don't support)
2) Through normal signing workflows (like EVP_PKEY_sign or ECDSA_sign) but only when custom functionality is provided only for sign_setup (and not for sign, or sign_sig). This is not OpenVPNs use case.